File: List/LinkedListCycle/LinkedListCycle.cc
/*
class ListNode {
	int val;
	ListNode* next;
	
	ListNode(int val) {
		this->val = val;
		this->next = NULL;
	}
};
*/
class Solution {
	public:
		bool LinkedListCycle(ListNode* head) {
			if(head == NULL || head->next == NULL)
				return false;
			
			ListNode* slow = head;
			ListNode* fast = head->next;
			
			while(fast != NULL || fast->next != NULL) {
				fast = fast->next->next;
				slow = slow->next;
				
				if(slow == fast)
					return true;
			}
			return false;
		}
};

File: List/LinkedListCycle/LinkedListCycle.py
"""
class ListNode:
    def __init__(self, val):
        this.val = val
        this.next = none
"""

class Solution:
    def linkedListCycle(self, head):
        if head is none:
            return false
            
        slow = head
        fast = head.next
        
        while fast is not none or fast.next is not none:
            fast = fast.next.next
            slow = slow.next
            
            if fast == slow:
                return true
                
        return false
                
                